core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -959,6 +976,12 @@ oppia.directive('conversationSkin', [
               $scope.pendingCardWasSeenBefore = false;
             }
             $scope.answerIsCorrect = false;
+            if ($scope.moveToExploration) {
```

Comment:
Again shouldn't all this be hidden within ExplorationPlayerStateService? Conversation skin directive shouldn't know or care about what mode we're in, it should just receive instructions to add a new card to the list of cards.